rln/src/lib.rs

```diff
@@ -193,18 +192,33 @@ mod test {
             rln_witness_from_values(identity_secret, &merkle_proof, x, epoch, rln_identifier);
 
         // We generate all relevant keys
-        let proving_key = &ZKEY();
-        let verification_key = &VK();
+        let proving_key = ZKEY().unwrap();
+        let verification_key = VK().unwrap();
         let builder = CIRCOM();
 
         // Let's generate a zkSNARK proof
-        let proof = generate_proof(builder, proving_key, &rln_witness).unwrap();
+        let proof = generate_proof(builder, &proving_key, &rln_witness).unwrap();
 
         let proof_values = proof_values_from_witness(&rln_witness);
 
         // Let's verify the proof
-        let success = verify_proof(verification_key, proof, &proof_values).unwrap();
+        let success = verify_proof(&verification_key, &proof, &proof_values).unwrap();
 
         assert!(success);
     }
+
+    #[test]
+    fn test_serialization() {
+        // We test witness serialization
+        let rln_witness = rln_witness_from_json(WITNESS_JSON);
```

Comment:
this is fine, but I don't think we'll use the json? That was there as an example, it isn't part of the C API usage through nwaku (e.g.)

Reply:
No, indeed we don't use it right now. But was useful in the debugging phase, so we might keep it in the codebase. Also in the future JSON may help exchanging data from different architecture/endpoints.

Reply:
Ok agree, let's keep it in for testing and possibly future APIs